pkg/server/scraper_benchmark_test.go

```diff
@@ -0,0 +1,311 @@
+// Copyright 2022 The Kubernetes Authors.
```

Comment:
I've added a benchmark test here, but not sure if it's correct. Because the benchmark test in the master and this pr cannot be consistent. Are the test results reasonable? 
/cc @serathius

Reply:
Purpose of benchmark was to compare performance cost of new vs old approach. If we cannot make a consistent benchmark between main branch and this PR then it doesn't makes sense. Please first develop a benchmark for the main branch.

Reply:
ok, sounds reasonable. I will implement it in the main branch firstly